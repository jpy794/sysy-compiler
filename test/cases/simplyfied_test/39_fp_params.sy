int k;
float params_f40(float x0, float x1, float x2, float x3, float x4, float x5,
                 float x6, float x7, float x8, float x9, float x10, float x11,
                 float x12, float x13, float x14, float x15, float x16,
                 float x17, float x18, float x19, float x20, float x21,
                 float x22, float x23, float x24, float x25, float x26,
                 float x27, float x28, float x29, float x30, float x31,
                 float x32, float x33, float x34, float x35, float x36,
                 float x37, float x38, float x39) {
  if (x39 != 0) {
    float arr[10] = {x0 + x1 + x2 + x3,     x4 + x5 + x6 + x7,
                     x8 + x9 + x10 + x11,   x12 + x13 + x14 + x15,
                     x16 + x17 + x18 + x19, x20 + x21 + x22 + x23,
                     x24 + x25 + x26 + x27, x28 + x29 + x30 + x31,
                     x32 + x33 + x34 + x35, x36 + x37 + x38 + x39};
    putfarray(10, arr);
    return arr[k];
  } else {
    return params_f40(x1, x2, x3, x4, x5, x6, x7, x8, x9, x10, x11, x12, x13,
                      x14, x15, x16, x17, x18, x19, x20, x21, x22, x23, x24,
                      x25, x26, x27, x28, x29, x30, x31, x32, x33, x34, x35,
                      x36, x37, x38, x39, x0 + x1 + x2);
  }
}

int main() {
  float arr[40][3];

  k = getint();
  int i = 0;
  while (i < 40) {
    getfarray(arr[i]);
    i = i + 1;
  }

  params_f40(
      arr[0][k], arr[1][k], arr[2][k], arr[3][k], arr[4][k], arr[5][k],
      arr[6][k], arr[7][k], arr[8][k], arr[9][k], arr[10][k], arr[11][k],
      arr[12][k], arr[13][k], arr[14][k], arr[15][k], arr[16][k], arr[17][k],
      arr[18][k], arr[19][k], arr[20][k], arr[21][k], arr[22][k], arr[23][k],
      arr[24][k], arr[25][k], arr[26][k], arr[27][k], arr[28][k], arr[29][k],
      arr[30][k], arr[31][k], arr[32][k], arr[33][k], arr[34][k], arr[35][k],
      arr[36][k], arr[37][k], arr[38][k], arr[39][k]);
  return 0;
}
