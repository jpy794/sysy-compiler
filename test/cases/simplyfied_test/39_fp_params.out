10: 0x1.aec9c2p+2 0x1.6187ep+2 0x1.6444fep+2 0x1.5dcf9ep+2 0x1.5d7f4cp+2 0x1.7ee688p+2 0x1.75af34p+2 0x1.870674p+2 0x1.75bafp+2 0x1.9b8604p+2
0
